src/Init/Data/BitVec/Bitblast.lean

```diff
@@ -429,6 +432,640 @@ theorem shiftLeft_eq_shiftLeftRec (x : BitVec w₁) (y : BitVec w₂) :
   · simp [of_length_zero]
   · simp [shiftLeftRec_eq]
 
+/- # udiv/urem recurrence for bitblasting -/
+
+/-
+r = n - d * q
+r = n - d * (∑ i, 2^i * q.getLsb i)
+
+-/
+
+
+/-!
+Let us study an instructive counterexample to the claim that
+  `n = d * q + r` for (`0 ≤ r < d`) uniquely determining q and r *over bitvectors*.
+
+- Let `bitwidth = 3`
+- Let `n = 0, d = 3`
+- If we choose `q = 2, r = 2`, then d * q + r = 6 + 2 = 8 ≃ 0 (mod 8) so satisfies.
+- But see that `q = 0, r = 0` also satisfies, as 0 * 3 + 0 = 0.
+- So for (`n = 0, d = 3`), both:
+    `q = 2, r = 2` as well as
+    `q = 0, r = 0` are solutions!
+
+It's easy to cook up such examples, by chosing `(q, r)` for a fixed `(d, n)`
+such that `(d * q + r)` overflows.
+-/
```

Comment:
It wouldn't hurt to have a bit more context to this counter-example, why do we care?